src/Microsoft.AspNet.OData.Shared/Query/SkipTokenQueryOption.cs

```diff
@@ -0,0 +1,352 @@
+        {
+            if (validationSettings == null)
+            {
+                throw Error.ArgumentNull("validationSettings");
+            }
+
+            if (Validator != null)
+            {
+                Validator.Validate(this, validationSettings);
+            }
+        }
+
+        private IQueryable ApplyToCore(IQueryable query, ODataQuerySettings querySettings, OrderByQueryOption orderBy)
+        {
+            if (Context.ElementClrType == null)
+            {
+                throw Error.NotSupported(SRResources.ApplyToOnUntypedQueryOption, "ApplyTo");
+            }
+            ExpressionBinderBase binder = new FilterBinder(Context.RequestContainer);
+            IDictionary<string, OrderByDirection> directionMap = PopulateDirections(orderBy);
+            bool parameterizeConstant = querySettings.EnableConstantParameterization;
+            ParameterExpression param = Expression.Parameter(Context.ElementClrType);
+            Expression where = null;
+            int count = 0;
+            /* We will create a where lambda of the following form -
+             * Where (true AND Prop1>Value1)
+             * OR (true AND Prop1=Value1 AND Prop2>Value2)
+             * OR (true AND Prop1=Value1 AND Prop2=Value2 AND Prop3>Value3)
+             * and so on...
+             * Adding the first true to simplify implementation.
+             */
+            Expression lastEquality = Expression.Constant(true);
+            foreach (KeyValuePair<string, object> item in _propertyValuePairs)
+            {
+                string key = item.Key;
+                MemberExpression property = Expression.Property(param, key);
+                object value = item.Value;
+
+                Expression compare = null;
+                ODataEnumValue enumValue = value as ODataEnumValue;
+                if (enumValue != null)
+                {
+                    value = enumValue.Value;
+                }
+                Expression constant = parameterizeConstant ? LinqParameterContainer.Parameterize(value.GetType(), value) : Expression.Constant(value);
+                if (directionMap.ContainsKey(key))
+                {
+                    compare = directionMap[key] == OrderByDirection.Descending ? binder.CreateBinaryExpression(BinaryOperatorKind.LessThan, property, constant, true) : binder.CreateBinaryExpression(BinaryOperatorKind.GreaterThan, property, constant, true);
+                }
+                else
+                {
+                    compare = binder.CreateBinaryExpression(BinaryOperatorKind.GreaterThan, property, constant, true);
+                }
+
+                Expression condition = Expression.AndAlso(lastEquality, compare);
+                where = where == null ? condition : Expression.OrElse(where, condition);
+
+                lastEquality = Expression.AndAlso(lastEquality, binder.CreateBinaryExpression(BinaryOperatorKind.Equal, property, constant, true));
+                count++;
```

Comment:
>count++; [](start = 16, length = 8)

what is count used for? I see it is initialized, then incremented in the loop, but don't see where its value is used.

Reply:
Leftover from prototype, deleted.